src/main/java/com/example/gymApp/service/TrainerService.java

```diff
@@ -42,6 +42,15 @@ public TrainerService(TrainerRepository trainerRepository, UserRepository userRe
     this.trainingRepository = trainingRepository;
   }
 
+  public Trainer saveTrainer(Trainer trainer){
+    return trainerRepository.save(trainer);
```

Comment:
You are return trainer, so better to return it without cache. To do this use `saveAndFlush`